File: frontend/src/app/types/analytics.ts
export interface AnalyticsPoint {
  timestamp: string;
  value: number;
}

export interface AnalyticsSensorData {
  sensor_id: number;
  sensor_name: string;
  unit?: string | null;
  points: AnalyticsPoint[];
}

export interface AnalyticsResponse {
  from: string;
  to: string;
  data: AnalyticsSensorData[];
}

File: backend/routes/response_models/analytics.py
from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel


class TimeSeriesPoint(BaseModel):
    timestamp: str
    value: float


class SensorTimeSeriesData(BaseModel):
    sensor_id: int
    sensor_name: str
    unit: str
    points: list[TimeSeriesPoint]
    min_value: Optional[float] = None
    max_value: Optional[float] = None
    avg_value: Optional[float] = None


class AnalyticsResponse(BaseModel):
    generated_at: datetime
    range_start: datetime
    range_end: datetime
    data: list[SensorTimeSeriesData]
